### ISSUE TITLE
Cannot run gns3 from a python3 virtual env

### ISSUE THREAD

**Issue by jean-christophe-manciot** [GNS3/gns3-gui#3311]
**Describe the bug**
Launching gns3 from the virtual env fails.

**GNS3 version and operating system (please complete the following information):**
 - Ubuntu 21.10
 - GNS3 version 2.2.31
 - Any use of the GNS3 VM or remote server (ESXi, bare metal etc.): no

**To Reproduce**
1. apt install python3.9-venv
2. mkdir -pv /opt/gns3
3. cd /opt/gns3
4. python3.9 -m venv venv
5. . /opt/gns3/venv/bin/activate
6. python3.9 -m pip install -I -U pip 
7. python3.9 -m pip install -I -U gns3-gui gns3-server PyQt6 sip
8. gns3
```
Fail update installation: No module named 'sip'
Can't import Qt modules: Qt and/or PyQt is probably not installed correctly...
```
No such issue when lauching gns3 outside of the venv.

**Comment by grossmj** [comment]
PyQt6 isn't supported. Install PyQt5 instead

**Comment by jean-christophe-manciot** [comment]
With PyQt5, I get:
```
Fail update installation: /opt/gns3/venv/lib/python3.9/site-packages/PyQt5/QtCore.abi3.so: undefined symbol: _ZdlPvm, version Qt_5
Can't import Qt modules: Qt and/or PyQt is probably not installed correctly...
```

**Comment by grossmj** [comment]
Are you still working on this? Have you tried to launch gns3 like this: `python3.9 -m gns3`?

**Comment by jean-christophe-manciot** [comment]
@grossmj 
> Are you still working on this?

I really need to be able to launch gns3 in a venv due to the incompatibilities between gui/server requirements and other pip packages when launched outside a venv

> Have you tried to launch gns3 like this: `python3.9 -m gns3`?
Samle issue despite PyQt5 installed inside the venv:
```
. /opt/gns3/venv/bin/activate
python3.9 -m gns3
Fail update installation: /opt/gns3/venv/lib/python3.9/site-packages/PyQt5/QtCore.abi3.so: undefined symbol: _ZdlPvm, version Qt_5
Can't import Qt modules: Qt and/or PyQt is probably not installed correctly...
```

**Comment by grossmj** [comment]
Looks like there is a conflict with something. Try to specify the PyQt5 version like this: `python -m pip install PyQt5==5.15.6` or scrap the virtual environment and try again from scratch.

**Comment by jean-christophe-manciot** [comment]
Same exact issue after trying from a clean slate and adding PyQt5==5.15.6.

Where is ```_ZdlPvm,``` supposed to come from?

**Comment by grossmj** [comment]
I am suspecting a conflict with the Qt libs installed via apt but I don't know how to solve this.

What do you have if you type `dpkg -l | grep PyQt5`?

**Comment by jean-christophe-manciot** [comment]
```
apt-cache policy python3-pyqt5
python3-pyqt5:
  Installed: 5.15.6+dfsg-1+b2
```
This package is required by other apt packages such as backintime-qt.

```
# apt show python3-pyqt5|grep Depends

Depends: python3-pyqt5.sip (<< 13), python3-pyqt5.sip (>= 12.8), python3 (>= 3~), python3:any, libc6 (>= 2.14), libgcc-s1 (>= 3.0), libpython3.10 (>= 3.10.0), libqt5core5a (>= 5.15.1), libqt5dbus5 (>= 5.14.1), libqt5designer5 (>= 5.9.1), libqt5gui5 (>= 5.1.0), libqt5gui5 (>= 5.15.1) | libqt5gui5-gles (>= 5.15.1), libqt5help5 (>= 5.15.1), libqt5network5 (>= 5.15.1), libqt5printsupport5 (>= 5.14.1), libqt5test5 (>= 5.11.0~rc1), libqt5widgets5 (>= 5.15.1), libqt5xml5 (>= 5.15.1), libstdc++6 (>= 5), qtbase-abi-5-15-2
```

How can there be a conflict since the apt Qt  libs are installed in /usr/lib?

I have found [a thread](https://github.com/Theano/Theano/issues/5141#issuecomment-256323855) which suggests a conflict with a too recent gcc.

```
gcc:
  Installed: 4:11.2.0-2
```

**Comment by cristian-ciobanu** [comment]
I saw this error on the Archlinux forum which also mentioned backintime application.
There might be a conflict between pip and apt packages and in that case the solution was to remove the pip package. Check here for details: https://bbs.archlinux.org/viewtopic.php?id=266419

**Comment by jean-christophe-manciot** [comment]
@cristian-ciobanu 
If I don't install PyQt5 inside the venv, I get another type of error:
```
. /opt/gns3/venv/bin/activate
python3.9 -m gns3
Fail update installation: No module named 'sip'
Can't import Qt modules: Qt and/or PyQt is probably not installed correctly...
```

**Comment by b-ehlers** [comment]
I just tested with Ubuntu 22.04 and I had no issues installing and running GNS3 in a venv with QT5 provided by Ubuntu.

What I did is very similar to yours:

1. sudo apt-get install python3-pyqt5 python3-pyqt5.qtsvg python3-pyqt5.qtwebsockets
2. sudo apt-get install python3-venv
3. mkdir -pv /opt/gns3
4. cd /opt/gns3
5. python3 -m venv --system-site-packages venv
6. . /opt/gns3/venv/bin/activate
7. python3 -m pip install -I -U gns3-gui gns3-server

Did you add the option `--system-site-packages`, when creating the venv?  This is necessary, when you want to use the system provided modules in a venv.
